src/cloud_provider/aws/kubernetes/mod.rs

```diff
@@ -119,1675 +109,1168 @@ pub struct Options {
-        {
-            return Err(EngineError::new_cannot_copy_files_from_one_directory_to_another(
-                event_details,
-                bootstrap_charts_dir,
-                common_charts_temp_dir,
-                e,
-            ));
+            skip_kubernetes_step = true;
+            "".to_string()
         }
+    };
+
+    // should apply before destroy to be sure destroy will compute on all resources
+    // don't exit on failure, it can happen if we resume a destroy process
+    let message = format!(
+        "Ensuring everything is up to date before deleting cluster {}/{}",
+        kubernetes.name(),
+        kubernetes.id()
+    );
+
+    kubernetes.send_to_customer(&message, &listeners_helper);
+    kubernetes
+        .logger()
+        .log(EngineEvent::Info(event_details.clone(), EventMessage::new_from_safe(message)));
+
+    kubernetes.logger().log(EngineEvent::Info(
+        event_details.clone(),
+        EventMessage::new_from_safe("Running Terraform apply before running a delete.".to_string()),
+    ));
+
+    if let Err(e) = cmd::terraform::terraform_init_validate_plan_apply(temp_dir.as_str(), false) {
+        // An issue occurred during the apply before destroy of Terraform, it may be expected if you're resuming a destroy
+        kubernetes.logger().log(EngineEvent::Error(
+            EngineError::new_terraform_error_while_executing_pipeline(event_details.clone(), e),
+            None,
+        ));
+    };
 
-        let kubernetes_config_file_path = match self.get_kubeconfig_file_path() {
-            Ok(x) => x,
-            Err(e) => {
-                let safe_message = "Skipping Kubernetes uninstall because it can't be reached.";
-                self.logger().log(EngineEvent::Warning(
-                    event_details.clone(),
-                    EventMessage::new(safe_message.to_string(), Some(e.message(ErrorMessageVerbosity::FullDetails))),
-                ));
-
-                skip_kubernetes_step = true;
-                "".to_string()
-            }
-        };
-
-        // should apply before destroy to be sure destroy will compute on all resources
-        // don't exit on failure, it can happen if we resume a destroy process
+    if !skip_kubernetes_step {
+        // should make the diff between all namespaces and qovery managed namespaces
         let message = format!(
-            "Ensuring everything is up to date before deleting cluster {}/{}",
-            self.name(),
-            self.id()
+            "Deleting all non-Qovery deployed applications and dependencies for cluster {}/{}",
+            kubernetes.name(),
+            kubernetes.id()
         );
-        self.send_to_customer(&message, &listeners_helper);
-        self.logger()
-            .log(EngineEvent::Info(event_details.clone(), EventMessage::new_from_safe(message)));
 
-        self.logger().log(EngineEvent::Info(
+        kubernetes.logger().log(EngineEvent::Info(
```

Comment:
OPT: no need to add extra info:

```
kubernetes.logger().log(EngineEvent::Info(
            event_details.clone(),
            EventMessage::new_from_safe("Deleting all non-Qovery deployed applications and dependencies for cluster".to_string()),
        ));
```

In this case I think it's not important since this message is reused to be sent to old logging system, but just to keep in mind.

Reply:
FAR